core/src/main/scala/org/apache/spark/deploy/history/HistoryServer.scala

```diff
@@ -179,7 +180,11 @@ class HistoryServer(
   }
```

Comment:
Finally I found this approach also have some issues. Both live UI and history UI depend on /api/v1/applications endpoint which uses UIRoot class to get the app list. UIRoot has two subclasses: SparkUI and HistoryServer. SparkUI's getApplicationList() returns iterator of ApplicationInfo directly, and HistoryServer's getApplicationList() needs to map ApplicationHistoryInfo to ApplicationInfo for the whole list which is the most time-consuming part. So it is unavoidable to do the transformation for the whole list if we want to return Iterator[ApplicationInfo] from getApplicationList(). I can add a method to return Iterator[ApplicationHistoryInfo] in UIRoot class, but it makes no sense for SparkUI to implement this irrelevant method. Finally I decided to add back the getApplicationListView() method.